### PR TITLE
Add extra resource limits to ulimit

### PR DESCRIPTION
**Pull request by zanchey**
I've been through the manual pages for Linux, and added most of the the resource
limits that aren't already available. I skipped `RLIMIT_LOCKS` because it only
applies to long-obsolete versions of the kernel.

Fixes issue #8786.

## TODOs:

- [x] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: src/builtins/ulimit.cpp [hunk 1]
```diff
@@ -200,10 +220,22 @@ maybe_t<int> builtin_ulimit(parser_t &parser, io_streams_t &streams, const wchar
...
+                what = RLIMIT_NICE;
+                break;
+            }
+#endif
             case 'f': {
                 what = RLIMIT_FSIZE;
                 break;
             }
+#ifdef RLIMIT_SIGPENDING
+            case 'i': {
```
Comment (faho) [inline src/builtins/ulimit.cpp:234]: I'm assuming these option letters are just taken from bash?

Comment (zanchey) [reply src/builtins/ulimit.cpp:234]: Most of them aren't in bash, so I've taken them from [prlimit(1)](https://man7.org/linux/man-pages/man1/prlimit.1.html).

File: src/builtins/ulimit.cpp [hunk 2]
```diff
@@ -200,10 +220,22 @@ maybe_t<int> builtin_ulimit(parser_t &parser, io_streams_t &streams, const wchar
                 what = RLIMIT_DATA;
                 break;
             }
+#ifdef RLIMIT_NICE
```
[the remark this answers is not in the document]
Comment (zanchey) [reply src/builtins/ulimit.cpp:223]: Maybe set the `what` value to -1 if the option is not defined, and print an error further down?

These are all on various versions of Linux. There are a bunch more on FreeBSD and NetBSD, which I can add. OpenBSD and macOS have a couple of extras too, but they're a subset of the others. Solaris has some too.

**Comment by zanchey** [comment]
I've included the macros from NetBSD and FreeBSD, added some basic tests, updated the completions and the documentation, and made the utility return a specific error if the option is valid but not supported.

**Review by faho — APPROVED** [review]
LGTM!

**Pull request merged**